compiler/rustc_typeck/src/check/method/probe.rs

```diff
@@ -262,6 +262,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             "probe(self_ty={:?}, return_type={}, scope_expr_id={})",
             self_ty, return_type, scope_expr_id
         );
+        self.set_tainted_by_errors();
```

Comment:
@estebank I moved it here because we only hit it when we're in an error reporting context here as `probe_for_return_type` is only used to generate suggestions whereas in the old location we could hit the `set_tainted_by_errors` call when we weren't doing error reporting

Reply:
If you look at the code in `demand_coerce_diag` once we move after the `try_coerce` we are always in an error reporting pathway.